### PR TITLE
ISPN-2145 Properly log JGroups initialization/configuration failures and make sure they don't get swallowed

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-2145

Cleanly cherry-picks to 7.2.x

### CODE REVIEW

File: core/src/test/java/org/infinispan/remoting/jgroups/NonExistingJGroupsConfigTest.java [hunk 1]
```diff
@@ -1,42 +1,66 @@
...
+         "<cache-container default-cache=\"default\">" +
+         "   <jmx domain=\"NonExistingJGroupsConfigTest_channelLookupTest\" />\n" +
+         "   <transport stack=\"dummy\" cluster=\"demoCluster\" />\n" +
+         "   <replicated-cache name=\"default\" />\n" +
+         "</cache-container>" +
+         TestingUtil.INFINISPAN_END_TAG;
+      EmbeddedCacheManager cm = null;
+      try {
+         cm = new DefaultCacheManager(new ByteArrayInputStream(config.getBytes()));
+         cm.getCache();
```
Comment (Sanne) [inline core/src/test/java/org/infinispan/remoting/jgroups/NonExistingJGroupsConfigTest.java]: just a thought.. isn't it wrong that this requires the getCache() invocation for the user to actually realize its mistake?

**Comment by Sanne** [comment]
It's now merged in master. @tristantarrant could you check the JIRA version and if you need it in other branches? I'm not sure about master vs 8.0 vs 7.2 state

**Comment by tristantarrant** [comment]
I pushed to 7.2.x as well

File: core/src/test/java/org/infinispan/remoting/jgroups/NonExistingJGroupsConfigTest.java (hunk 1 above)
Comment (tristantarrant) [inline core/src/test/java/org/infinispan/remoting/jgroups/NonExistingJGroupsConfigTest.java]: The transport only starts lazily

Comment (Sanne) [inline core/src/test/java/org/infinispan/remoting/jgroups/NonExistingJGroupsConfigTest.java]: I know that :-) but explaining how our implementation works is a poor excuse for a usability problem.
At least validation of configuration files should be complete when the CacheManager is started.

**Pull request closed without merging**